Add test coverage for oauth

MattMSumner opened the pull request:
Why:

- We'd like to have test coverage to sanity check any changes that we
make to the library.

This change addresses the need by:

- Adding test coverage for getAccessToken and refreshAccessToken.

NOTE: Since oauth requires a server to do the handshake dance with
HubSpot these tests will not pass with NOCK_OFF=true. I'm not sure if we
should disable these tests in that case or come up with another
solution.

When writing these tests I wrote them without mocking but had to
hardcode client_id and client_secret then do half the oauth handshake to
get a valid code that I could then trade for a token.

SViccari commented on test/oauth.js:
```diff
@@ -23,41 +22,87 @@ describe('oauth', function() {
+      it('should return a token from hubspot', function() {
+        if (process.env.NOCK_OFF) {
+          console.warn(
+            'we currently do not test oauth with NOCK_OFF set to true',
```
Nice 👍 Can this warning be combined with the above `if (process.env.NOCK_OFF)` statement?

MattMSumner replied:
Yes but it looks terrible in the console so I've removed it.

SViccari commented on test/oauth.js:
```diff
@@ -23,41 +22,87 @@ describe('oauth', function() {
+    fakeHubspotApi.setupServer({ postEndpoints: [oauthTokenEndpoint] })
+
+    if (process.env.NOCK_OFF) {
+      it('should return a token from hubspot')
```
Thoughts on adding a similar warning here?

MattMSumner replied:
Turns out it doesn't look great

SViccari reviewed (approved):
LGTM 🐘

MattMSumner committed:
Add test coverage for oauth

Why:

We'd like to have test coverage to sanity check any changes that we
make to the library.

This change addresses the need by:

Adding test coverage for getAccessToken and refreshAccessToken.

NOTE: Since oauth requires a server to do the handshake dance with
HubSpot these tests will not pass with NOCK_OFF=true. We therefore mark
the tests as pending when NOCK_OFF=true.
